src/Toml/Parser.hs

```diff
@@ -80,7 +80,7 @@ basicStringP :: Parser Text
 basicStringP = lexeme $ Text.pack <$> (char '"' *> anyChar `manyTill` char '"')
 
 textP :: Parser Text
-textP = multilineStringP <|> multilineLiteralStringP <|> literalStringP <|> basicStringP
+textP = multilineBasicStringP <|> multilineLiteralStringP <|> literalStringP <|> basicStringP
```

Comment:
All these parsers are not backtracking. Which means that if you try to parse string like this:

```haskell
"foo"
```

It will try `multilineBasicStringP` first. This parser will consume single `"` and will fail because here `quotes >> optional eol` parser `quotes` is not under `try`. So my intuition that `textP` will fail on `"foo"` unless you add `try` to `quotes` parser. But if this currently works then it's a mystery to me 🤷‍♂️

Reply:
You can read the note here about [`try`](https://hackage.haskell.org/package/megaparsec-6.5.0/docs/Text-Megaparsec.html#v:try):

It says:
>Please note that as of Megaparsec 4.4.0, string backtracks automatically (see tokens), so it does not need try.

I'm using `string` in both `multilineBasicStringP` and `multilineLiteralStringP` and I'm counting on it to backtrack.

Reply:
Ah, okay, `string` parser is an exception.